### PR TITLE
Cabal cleanup feat. new-* commands

### PR DESCRIPTION
**Pull request by fxdpntthm**
from now on you can run:
   - `make alb` to build alb (also pulls in the dependencies)
   - `make ilab` to build ilab (also pulls in the dependencies)
   - `make tests-alb` to run the tests to check sanity of alb compiler
   - `make tests-ilab` to run the tests and check sanity of ilab solver
   - `make install-alb` run tests and dist the alb artifacts
   - `make install-ilab` run tests and dist the ialb artifacts

All the make tasks essentially run cabal new-* commands

Also README styling because its 2019 and markdown looks a little cooler than ascii

### CODE REVIEW

**Comment by peschkaj** [comment]
:+1:

The only error I see happens during tests running when I get:

```
Testing tests/example-test.out: 
testHarness.hs: tests/example-test.out: rawSystem: runInteractiveProcess: exec: does not exist (No such file or directory)
tests/Makefile:108: recipe for target 'tests-alb' failed
make: *** [tests-alb] Error 1
```

**Review by peschkaj — APPROVED** [review]
Looks good 👍

**Comment by fxdpntthm** [comment]
So the way the tests run is:

alb first spits out the *.out file using *.hb
testHarness.hs runs it using the process.

The error that you see is because `tests/example-test.out` was not generated 
Is it the case that you see `test/example-test` generated?

Apoorv

**Comment by peschkaj** [comment]
Ah, I see why this is failing now: `tests/Makefile` still has `MIL_DIR=$(HOME)/workspace/habit-lang/mil-tools` which is not where mil-tools lives locally. 

The install path for mil-tools is in a `bin` off of whichever directory contains `build.xml`. I suspect on your machine, this is in `~/workspace/habit-lang/mil-tools/`, over here it's in `~/src/habit-lang/mil-tools`. 

Once I change `MIL_DIR` to `../mil-tools` all of the tests to run and pass.

I have pushed that change, but I'd be happy with any better solution.

File: tests/Makefile [hunk 1]
```diff
@@ -90,7 +90,7 @@ ALB_DIR=$(pwd)
 ALB_TEST_DIR=tests
 
 # MIL locations
-MIL_DIR=$(HOME)/workspace/habit-lang/mil-tools
```
Comment (fxdpntthm) [inline tests/Makefile]: This needs a better way..

We can have everyone having different relative paths of their mil-tools and alb compiler
Is there any generic way of solving this problem of relative paths?

**Pull request merged**